### ISSUE TITLE
Contact form: cannot insert a form using do_shortcode

### ISSUE THREAD

**Issue by jeherve** [Automattic/jetpack#102]
Steps to reproduce:
1. Edit one of your theme's template files and add a Contact form with the do_shortcode function:

``` php
<?php echo do_shortcode( "[contact-form][contact-field label='Name' type='name' required='1'/][contact-field label='Email' type='email' required='1'/][contact-field label='Website' type='url'/][contact-field label='Comment' type='textarea' required='1'/][/contact-form] " ); ?>
```
1. Load a page using this template file: the contact form is displayed.
2. Fill in the form, and submit it
3. The form is not submitted properly.

Reported in this trac ticket:
https://plugins.trac.wordpress.org/ticket/1762

**Comment by clarklab** [comment]
For me, the form actually sends just fine, it's just sending to the wrong address. Using do_shortcode somehow makes it ignore the "to" parameter.

The thing that has me really puzzled is how my address keeps getting the submissions. We've changed the page author, the admin contact, my user account email: I simply have no idea how it's getting my email.

**Comment by lstude** [comment]
As an update to @clarklab's previous comment -- we were able to use the following workaround to get our footer form to send to the proper address:

1. Update user's email address to wherever you want form to send: ![update-user](http://content.screencast.com/users/lstude/folders/Snagit/media/7c028f17-83ee-4841-a12f-4499100604c8/2015-05-22_13-26-29.png)
2. Change author on whichever page is tied to your main form to the updated user (in our case, it was Home - Front Page ![change author](http://content.screencast.com/users/lstude/folders/Snagit/media/18a770b2-fd42-4d88-9a4b-95bc183f4095/2015-05-22_13-28-23.png)
3. Form now submits to proper address

**Comment by RobertShippey** [comment]
@lstude I think your workaround might help me here, but my contact form is on the single template for a custom post type, so do you think I would have to change the author of every item of my custom post type to make the form work across all instances of it?

I've found the same as @clarklab, the `do_shortcode` function seems to make the form ignore the `to` parameter. Anyone else found any other ways to fix this?

**Comment by Forket** [comment]
Also i've found that if shortcode is created in page editor - then everything is OK: feedback items created, email sent.
BUT if I use same shortcode in php file - form saves, everything is ok, except sending notification to email :-\

```
<?php
			    echo do_shortcode("[contact-form to='alex@*****.com' subject='teehljjkhfksdjhkl'][contact-field label='Name' type='name' required='1'/][contact-field label='Email' type='email' required='1'/][contact-field label='Website' type='url'/][contact-field label='Comment' type='textarea' required='1'/][/contact-form]");
			    ?>
```

**Comment by jeherve** [comment]
@Forket that seems to be a different issue, and I wasn't able to reproduce.

Could you check your server's email logs to make sure the email wasn't blocked when you sent it out.

Could you also try to use a plugin like [Postman](https://wordpress.org/plugins/postman-smtp/) to send all emails through a different email provider, and see if it helps?

Thanks!

**Comment by lamdayap** [comment]
Also reported here: #2678300-t

**Comment by brbrr** [comment]
The issue is still there. Jetpack 6.3-beta

**Comment by htdat** [comment]
Still the issue. Just tested with 7.9-alpha.

**Comment by simison** [comment]
@jeherve I think we can close this issue at this point, as a wont-fix? One can use `do_blocks()` and modern block markup if there's a need to do something like this programmatically.

**Comment by jeherve** [comment]
Yes, I think it's a lot less relevant in a world of block-based widgets and the Form block being available to insert in templates in block-based themes.
